Feature samples selection

adityaekawade opened the pull request.

chmille4 commented on app/dataSelect.js:
```diff
@@ -94,7 +95,9 @@ DataSelect.prototype.addBuildListener = function() {
 				return;
 			}
 			genomeBuildHelper.setCurrentBuild(value);
+			//console.log("build val", value);
```
Can remove this debug statement

adityaekawade replied:
Hi, I have removed all the debug statements which were left in the latest commit.

chmille4 commented:
Lot of changes, so I didn't go through it with a fine tooth comb, but it looks good to me. I tested it as well and it looks good, but @tonydisera could you test it as well just to make sure. Then we should be able to merge

chmille4 commented:
Actually just found a bug. If you specify the .tbi file then two sampling events are fired. If you click the link below you can see the read sampled number go higher then lower then higher multiple times indicating two sampling commands have been executed.

http://localhost:3000/?species=Human&build=GRCh37&vcf=http%3A%2F%2Fs3.amazonaws.com%2Fvcf.files%2FExAC.r0.2.sites.vep.vcf.gz&tbi=http%3A%2F%2Fs3.amazonaws.com%2Fvcf.files%2FExAC.r0.2.sites.vep.vcf.gz.tbi

tonydisera commented:
I did some testing this morning.  Overall, it is working well.  Here are a few minor issues:
1.  When the vcf is specified in the URL, vcf.iobio loads correctly.  However, the 'filter samples' button doesn't appear.
2.  When the .tbi file is in a different URL, the 'Loading...' message shows before I've had a chance to enter the tbi URL.
<img width="1137" alt="screen shot 2018-01-26 at 8 34 26 am" src="https://user-images.githubusercontent.com/8420103/35447036-cd6ed08e-0273-11e8-92bc-f1f8169cb3cf.png">

adityaekawade commented:
@tonydisera :  Regarding (1), the filter sample button would appear only if the URL contains the samples. If it does not contain the samples, it would show just the load button. 
I am attaching the screenshots of both the cases. 

<img width="1159" alt="screen shot 2018-01-26 at 9 35 25 am" src="https://user-images.githubusercontent.com/16284713/35450322-e50f5dc2-027c-11e8-9fa2-5a6bbebf54e3.png">
<img width="1196" alt="screen shot 2018-01-26 at 9 35 48 am" src="https://user-images.githubusercontent.com/16284713/35450323-e5221fde-027c-11e8-95b4-461c943bfb72.png">

tonydisera commented:
Aditya,
I am referring to the use case when the VCF URL is specified in the launch url.  When vcf.iobio is launched, the 'Filter samples' doesn't appear even though the vcf url is a multi-sample vcf:
http://localhost/vcf.iobio.io/?vcf=https%3A%2F%2Fs3.amazonaws.com%2Fiobio%2Fsamples%2Fvcf%2Fplatinum-exome.vcf.gz&tbi=&samples=NA12878&build=GRCh37

adityaekawade commented:
Ok. Got it! I noticed the same thing now even when the page is refreshed. Thank you.

adityaekawade commented:
I have fixed these issues in the latest commit: 

1. Fixed the bug of two sampling commands fired when the tbi URL is specified. The read sampled number does not go higher-lower-higher now. 
2. The loading message appears only when both the URLs are entered. 
3. The filter sample button is now displayed when the vcf is specified in the url or when the page is refreshed. 

@tonydisera  @chmille4

tonydisera commented:
Hi Aditya -

It looks like this fix may have broken something.  When I enter a multi-sample vcf (from the landing page), the sample select dropdown no longer appears.  
<img width="779" alt="screen shot 2018-01-29 at 9 06 24 am" src="https://user-images.githubusercontent.com/8420103/35520324-f1a02068-04d3-11e8-9698-5941dfa9a8c3.png">


After clicking on load, the 'Filter samples' button does appear on the main page.



Launching vcf.iobio with the VCF URL now shows the 'Filter samples' button.
<img width="1349" alt="screen shot 2018-01-29 at 9 08 34 am" src="https://user-images.githubusercontent.com/8420103/35520412-31b934fa-04d4-11e8-8d6e-2da706bd99f2.png">

adityaekawade commented:
Hi Tony, 
Can you please confirm again, it is working fine for me. 

<img width="1278" alt="screen shot 2018-01-29 at 9 26 31 am" src="https://user-images.githubusercontent.com/16284713/35521394-d8ef27be-04d6-11e8-8d5f-8908508568d4.png">

tonydisera commented:
Oops!  You are right.  It is fixed.  Looks good.  You can merge to gh-pages (and master) to deploy to production.

tonydisera commented:
I just accepted this pull request and merged to gh-pages.  Aditya, can you please confirm that everything is working on production?  Thanks!

adityaekawade commented:
Yes, I am checking that! Thank you.

On 29 January 2018 at 10:57, Tony Di Sera <notifications@github.com> wrote:

> I just accepted this pull request and merged to gh-pages. Aditya, can you
> please confirm that everything is working on production? Thanks!
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tonydisera/vcf.iobio.io/pull/59#issuecomment-361330348>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APh8KSUwR74bQc2hRQU71Xk4IcWoaCE0ks5tPgaBgaJpZM4Rr1Zw>
> .
>

adityaekawade commented:
It's working fine on production!
